app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -55,24 +55,42 @@ def validate_authentication_status
       {:available => true, :message => nil}
     end
 
+    def self.verify_ssl_mode(security_protocol)
+      case security_protocol
+      when nil, '', 'non-ssl', 'ssl-without-validation'
```

Comment:
`nil, ''` are for treating existing providers as `non-ssl`, right?  Are you planning to migrate existing providers to have `non-ssl`?
And for 'non-ssl' this value doesn't matter, only using HTTP does, right?
It seems then the handling of nil, '', 'non-ssl' better belongs in `entrypoint()` when deciding HTTP/HTTPS.